de.tudarmstadt.ukp.dkpro.argumentation.nio/src/main/java/de/tudarmstadt/ukp/math/Sparse3DObjectMatrix.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) the Department of Informatics, Technische Universität Darmstadt. All Rights Reserved.
+ *
+ * Unauthorized distribution of this file via any medium is strictly prohibited.
+ */
+package de.tudarmstadt.ukp.math;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+
+/**
+ * TODO: This class is an ugly, quick hack: Replace this with an object matrix
+ * implementation from the library ultimately used for clustering in order to
+ * reduce the amount of code which needs to be written.
+ *
+ * @deprecated This class is an ugly, quick hack and should be replaced with an
+ *             object matrix implementation from the library ultimately used for
+ *             clustering in order to reduce the amount of code which needs to
+ *             be written.
+ * @author <a href="mailto:shore@ukp.informatik.tu-darmstadt.de">Todd Shore</a>
+ * @since Apr 29, 2016
+ *
+ */
+@Deprecated
```

Comment:
I can replace this with something "non-hacky" but it will take a very long time and, anyway, this particular class is only used in `de.tudarmstadt.ukp.dkpro.argumentation.annotations.SpanAnnotationMatrices` and `de.tudarmstadt.ukp.dkpro.argumentation.nio.writer.json.JCasTextSpanAnnotationGraphFactory`-- both of which are package-private.

Personally, I think this class' restricted usage combined with its deprecated annotation is enough to keep people from re-using it, and (not to sound arrogant) my time is probably better invested in things other than making all code perfect. If this is still unacceptable, I could remove all of `de.tudarmstadt.ukp.dkpro.argumentation.nio` from this repository and turn it into a separate, unpublished project which nevertheless relies on the *dkpro-argumentation* project. However, that would also take a decent amount of time.